src/core/Akka.Persistence/Eventsourced.Recovery.cs

```diff
@@ -39,9 +40,48 @@ public EventsourcedState(string name, bool isRecoveryRunning, StateReceive state
     public abstract partial class Eventsourced
     {
         /// <summary>
-        /// Processes a loaded snapshot, if any. A loaded snapshot is offered with a <see cref="SnapshotOffer"/> 
-        /// message to the actor's <see cref="ReceiveRecover"/>. Then initiates a message replay, either starting 
-        /// from the loaded snapshot or from scratch, and switches to <see cref="ReplayStarted"/> state. 
+        /// TBD
+        /// </summary>
+        /// <param name="action">TBD</param>
+        protected void RunTask(Func<Task> action)
+        {
+            Func<Task> wrap = () =>
+            {
+                Task t = action();
+                if (!t.IsCompleted)
+                {
+                    var tcs = new TaskCompletionSource<object>();
+                    t.ContinueWith(r =>
```

Comment:
You probably won't need `TaskCompletionSource` here. `ContinueWith` will return continuation task, while `TaskContinuationOption.NotOnFaulted | TaskContinuationOption.NotOnCancelled` should allow you to get rid of forwarding cancelations and faults.

There's also one thing to remember - things like actor's `Context` or `Self` are fragile and won't work in task continuations. This also concerns all methods invoked here.